### PR TITLE
fix: show promotional code only on paid tickets & registration

### LINKED ISSUE #8071: Registration: Only show "Enter Promotional Code" when there are paid tickets
Currently if there are only free registrations there is nevertheless the option to enter a promotional code. But, registrations are already free. So, no need to for promotional codes. The promotional code option should only show up if there are paid registrations available.

Screenshot from 2021-11-11 20-52-06

### PR DESCRIPTION
**Pull request by progmatic-99**
Fixes #8071 
ss_1636895475

### CODE REVIEW

File: app/templates/components/public/ticket-list.hbs [hunk 1]
```diff
@@ -198,7 +198,6 @@
             {{#if this.device.isMobile}}
               <div class="ui hidden divider"></div>
             {{/if}}
-            <a href="#" {{action 'togglePromotionalCode'}}>{{t 'Enter promotional code'}}</a>
```
Comment (maze-runnar) [inline app/templates/components/public/ticket-list.hbs:201]: no need to remove it from the component. fix the function `hasOnlyFreeTickets`.

**Review by maze-runnar — CHANGES_REQUESTED** [review]

File: app/templates/components/public/ticket-list.hbs (hunk 1 above)
Comment (progmatic-99) [reply app/templates/components/public/ticket-list.hbs:201]: What is its use there? It just points to _#_

Comment (progmatic-99) [reply app/templates/components/public/ticket-list.hbs:201]: It should not be shown when there are only free tickets.

Comment (maze-runnar) [reply app/templates/components/public/ticket-list.hbs:201]: I am saying that keep action and toggle promotion in same component according to best coding practices. Otherwise writing working code isn't a big thing

**Comment by pc-beast** [comment]
@progmatic-99 please test once on corner cases

**Comment by progmatic-99** [comment]
@mariobehling This can be merged.

**Review by pc-beast — APPROVED** [review]

**Review by mariobehling — APPROVED** [review]
